bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -553,7 +557,7 @@ public boolean performReplaceAndFind(String findString, String replaceString) {
 	public boolean performSelectAndReplace(String findString, String replaceString) {
 		resetStatus();
 		if (!isFindStringSelected(findString)) {
-			performSearch(findString);
+			performSearch(findString, false);
```

Comment:
```suggestion
                        disableOption(SearchOptions.INCREMENTAL);
			performSearch(findString, false);
			enableOption(SearchOptions.INCREMENTAL);
```

Reply:
The design flaw is as follows: the logic know about search options (defined in an enum) and search operation (defined as methods). Now "incremental" is basically both: an option and an operation. You can activate the "incremental" option to say that an incremental base position for searching shall be used, and you can perform an incremental search, i.e., one that does not move to the next search result but update the current one. The latter are actually two operations: one performed by updating the search input string and one by performing an explicit search option (via pressing enter or a button) with different behaviors. These are two different things that are currently mixed up, as you can see at the fact that the incremental option needs to be changed just to make an operation behave differently and not because the user actually wanted to specify some different option.

I have already prepared a more holistic improvement, which completely removes all these necessities by hiding the responsibility for distinguishing these two cases in the logic, so no consumer will need to care about the different operation modes. But as mentioned, that's too invasive for the upcoming release as we are already after M3, so I will provide that improvement for 2024-12 M1.

I am not in favor of the proposed fix of temporarily disabling the incremental search option in the logic. It is more invasive, as it performs additional modification of the incremental base position. I would like to keep the impact as small as possible to be confident that we do not introduce any regressions at this point in time. Only reasons for me to integrate the proposed fix would be an additional benefit in terms of fixing some unintended behavior with the current fix that I have not found so far.